src/3d/qgs3dmapsettings.h

```diff
@@ -670,6 +670,19 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject, public QgsTemporalRangeObjec
      */
     void setIsDebugOverlayEnabled( bool debugOverlayEnabled );
 
+    /**
+     * Returns the 3D scene's 2D extent in project's CRS
+     * \since QGIS 3.30
+     */
+    QgsRectangle extent() const { return mExtent; };
+
+    /**
+     * Sets the 3D scene's 2D \a extent in project's CRS
+     * Terrain will only be generated within this extent and layer 3D data will only be loaded within this extent.
```

Comment:
Worth clarifying in docstring:
Is it required that setExtent() is called during initialization?
Is it valid to set empty extent to use full project extent?